mageFile.go

```diff
@@ -0,0 +1,139 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+func Lint() error {
+	if err := sh.RunV("golangci-lint", "run", "--timeout", "3m"); err != nil {
+		return err
+	}
+
+	if err := sh.RunV("go", "mod", "tidy"); err != nil {
+		return err
+	}
+
+	return sh.RunV("git", "diff", "--exit-code")
+}
+
+func Test() error {
+	return sh.RunV("go", "test", "./...", "-race")
+}
+
+func BuildLocal() error {
+	ldFlags := generateLDFlags()
+
+	fmt.Println(ldFlags)
+	return sh.RunV("go", "build", "-trimpath", "-ldflags", ldFlags, "-o", "apko", "./cmd/apko/main.go")
+}
+
+func KoApply() error {
+	os.Setenv("LDFLAGS", generateLDFlags())
+
+	return sh.RunV("ko", "apply", "-BRf", "config/")
+}
+
+// BuildImagesLocal build images locally and not push
+func BuildImagesLocal() error {
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	os.Setenv("LDFLAGS", generateLDFlags())
+
+	return sh.RunV("ko", "publish", "--base-import-paths", "--bare", "--local",
+		"--platform=linux/amd64", "--tags", gitVersion, "--tags", gitCommit, "--tags", "latest",
+		"chainguard.dev/apko/cmd/apko")
+}
+
+// BuildImages build the images and push
+func BuildSignImages() error {
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	os.Setenv("LDFLAGS", generateLDFlags())
+	os.Setenv("COSIGN_EXPERIMENTAL", "true")
+
+	if os.Getenv("KO_DOCKER_REPO") == "" {
+		return errors.New("missing KO_DOCKER_REPO environment variable")
+	}
+
+	digest, err := sh.Output("ko", "build", "--base-import-paths", "--bare",
```

Comment:
--base-... and --bare are mutually exclusive

Reply:
humm did not know and they not complain, which one we should use?

Reply:
I generally use the "base" one, but I used `--bare` for the release build so `KO_DOCKER_REPO` is exactly where the image goes.

Reply:
ok, thanks